src/broker/module.c

```diff
@@ -599,16 +600,23 @@ module_t *module_add (modhash_t *mh,
     }
     if (!(p->path = strdup (path))
         || !(p->rmmod = zlist_new ())
-        || !(p->subs = zlist_new ())
-        || !(p->name_buf = strdup (path))
-        || !(p->name = module_name_from_path (p->name_buf)))
+        || !(p->subs = zlist_new ()))
         goto nomem;
+    if (name) {
+        if (!(p->name = strdup (name)))
```

Comment:
Should this be
```c
    if (!(p->name_buf = strdup (name)))
        goto nomem;
    p->name = p->name_buf;
```

To avoid leaking `p->name`? At least I don't see where `p->name` is ever freed.

Reply:
Oooh, good catch!  In the final code, p->name might be a pointer into p->name_buf or might be a new allocation. So this is a leak.  WIll fix!

Reply:
Fixed and force pushed.